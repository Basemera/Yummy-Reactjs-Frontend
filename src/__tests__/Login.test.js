import React from 'react';
import { shallow, enzyme } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import Login from '../components/auth/login';

describe('Login component', () => {
  const login = jest.fn();
  const wrapper = shallow(<Login login={login} loggedIn={false} location={{}} />);

  it('renders properly without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('it renders state initially', () => {
    expect(wrapper.state().email_field).toEqual('');
    expect(wrapper.state().password_field).toEqual('');
  });
});
