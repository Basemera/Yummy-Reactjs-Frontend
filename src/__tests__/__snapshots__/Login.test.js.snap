// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders properly without crashing 1`] = `
<div
  className="background"
>
  <div
    id="logbar"
  >
    <div
      className="form-group"
    >
      <h1>
        LOGIN
      </h1>
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-group"
        >
          <span
            className="input-group-addon"
            id="email"
          >
            <i
              className="fa fa-envelope"
            />
          </span>
          <input
            aria-describedby="email"
            className="form-control"
            id="emailsignup"
            name="email_field"
            onChange={[Function]}
            placeholder="example@gmail.com"
            type="email"
            value=""
          />
        </div>
        <br />
        <div
          className="input-group"
        >
          <span
            className="input-group-addon"
            id="email"
          >
            <i
              className="fa fa-lock"
            />
          </span>
          <input
            className="form-control"
            name="password_field"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <br />
        <button
          className="btn btn-default"
          id="signin"
          name="signin"
          type="submit"
        >
          <i
            className="glyphicon glyphicon-log-in"
          >
            LOGIN
          </i>
        </button>
      </form>
      <div
        className="change_link"
      >
        Not a member yet ?
        <Link
          replace={false}
          to="/register"
        >
          <h4
            className="toregister"
          >
            Register Here
          </h4>
        </Link>
      </div>
    </div>
    <a
      data-target="#forgot_password"
      data-toggle="modal"
      href="#"
    >
      ForgotPassword?
    </a>
    <PasswordResetEmail />
  </div>
  <footer
    className="footer"
  >
    <p>
      Â© Hadijah 2018
    </p>
  </footer>
</div>
`;
